dashboard/test/models/peer_review_test.rb

```diff
@@ -478,9 +487,18 @@ class PeerReviewTest < ActiveSupport::TestCase
   test 'create_for_submission rolls back if there is an error' do
     track_progress @level_source.id
 
+    # Fake that reviews have been submitted
+    PeerReview.find_by(submitter: @user, level: @level, status: nil).update data: 'fake-peer-review', reviewer: (create :teacher)
+    PeerReview.find_by(submitter: @user, level: @level, status: 'escalated').update status: 'rejected', reviewer: @instructor, from_instructor: true
+
+    # Unsubmit the level so we can check behavior on resubmit
+    user_level = UserLevel.find_by(user: @user, level: @level, script: @script)
+    user_level.update(submitted: false)
+
     original_peer_reviews = PeerReview.where(level_source_id: @level_source.id)
-    PeerReview.stubs(:create!).raises(Exception, "Some error")
+    assert_equal 2, original_peer_reviews.count
 
+    PeerReview.stubs(:create!).raises(Exception, "Some error")
     assert_raises(Exception) do
       track_progress @level_source.id
     end
```

Comment:
Similar story for this test, as above.

Reply:
The new `submitted_or_resubmitted?` isn't sufficient to pass this test, but in this case I genuinely believe it was a test problem.  It was calling `track_progress` a second time simulating absolutely no change to the submitted reflection, in which case we shouldn't need to update peer reviews anyway.

I've updated this test case to behave more like the "'resubmitting for review should create new PeerReview objects" test above, actually changing the `level_source_id` on the second submission to simulate an actual update to the reflection needing review.  I've also generalized the query we're using to check that nothing changed to use `level_id` instead of `level_source_id` to make sure we're actually checking what we say we're checking here.